mods/taskbar-dock-animation.wh.cpp

```diff
@@ -0,0 +1,436 @@
+    g_settings.effectRadius = Wh_GetIntSetting(L"EffectRadius");
+    if (g_settings.effectRadius <= 0) g_settings.effectRadius = 100;
+}
+
+HMODULE GetTaskbarViewModuleHandle() {
+    HMODULE module = GetModuleHandle(L"Taskbar.View.dll");
+    if (!module) module = GetModuleHandle(L"ExplorerExtensions.dll");
+    return module;
+}
+
+bool HookTaskbarViewDllSymbols(HMODULE module) {
+     // Taskbar.View.dll
+     WindhawkUtils::SYMBOL_HOOK taskbarViewHooks[] = {
+        {
+            {
+                LR"(public: virtual int __cdecl winrt::impl::produce<struct winrt::Taskbar::implementation::TaskbarFrame,struct winrt::Windows::UI::Xaml::IFrameworkElementOverrides>::MeasureOverride(struct winrt::Windows::Foundation::Size,struct winrt::Windows::Foundation::Size *))"
+            },
+            &TaskbarFrame_MeasureOverride_Original,
+            TaskbarFrame_MeasureOverride_Hook,
+        },
+    };
+    if (!HookSymbols(module, taskbarViewHooks, ARRAYSIZE(taskbarViewHooks))) {
+        Wh_Log(L"DockAnimation: HookSymbols failed.");
+        return false;
+    }
+    Wh_Log(L"DockAnimation: HookSymbols succeeded for TaskbarFrame_MeasureOverride.");
+    return true;
+}
+
+using LoadLibraryExW_t = decltype(&LoadLibraryExW);
+LoadLibraryExW_t LoadLibraryExW_Original;
+
+// Lazy hook path (if Taskbar.View.dll loads later)
+HMODULE WINAPI LoadLibraryExW_Hook(LPCWSTR lpLibFileName, HANDLE hFile, DWORD dwFlags) {
+    HMODULE module = LoadLibraryExW_Original(lpLibFileName, hFile, dwFlags);
+    if (!module) return module;
+
+    if (!g_taskbarViewDllLoaded && GetTaskbarViewModuleHandle() == module) {
+        if (!g_taskbarViewDllLoaded.exchange(true)) {
+            Wh_Log(L"DockAnimation: Taskbar.View.dll loaded, hooking symbols...");
+            if (HookTaskbarViewDllSymbols(module)) {
+                Wh_ApplyHookOperations();
+                Wh_Log(L"DockAnimation: Hooks applied.");
+            }
+        }
+    }
+    return module;
+}
+
+// Windhawk entry points
+BOOL Wh_ModInit() {
+    Wh_Log(L"DockAnimation: Wh_ModInit");
+    LoadSettings();
+
+    if (HMODULE taskbarViewModule = GetTaskbarViewModuleHandle()) {
+        // Fast path (module already loaded)
+        g_taskbarViewDllLoaded = true;
+        if (!HookTaskbarViewDllSymbols(taskbarViewModule)) return FALSE;
+        g_fastPathHooksApplied = false;
+    } else {
+        // Lazy path (wait for LoadLibrary of Taskbar.View.dll)
+        HMODULE kernelBaseModule = GetModuleHandle(L"kernelbase.dll");
+        auto pKernelBaseLibraryExW = (decltype(&LoadLibraryExW))GetProcAddress(kernelBaseModule, "LoadLibraryExW");
+        WindhawkUtils::SetFunctionHook(
+            pKernelBaseLibraryExW,
+            LoadLibraryExW_Hook,
+            &LoadLibraryExW_Original
+        );
+    }
+    return TRUE;
+}
+
+void Wh_ModAfterInit() {
+    Wh_Log(L"DockAnimation: Wh_ModAfterInit");
+    if (g_taskbarViewDllLoaded && !g_fastPathHooksApplied) {

```

Comment:
What is `g_fastPathHooksApplied` useful for? It's always `false` here, and it's never accessed afterwards.